src/language/typescript/tsWorker.ts

```diff
@@ -63,8 +65,35 @@ export class TypeScriptWorker implements ts.LanguageServiceHost, ITypeScriptWork
 	}
 
 	getScriptFileNames(): string[] {
-		const allModels = this._ctx.getMirrorModels().map((model) => model.uri);
-		const models = allModels.filter((uri) => !fileNameIsLib(uri)).map((uri) => uri.toString());
+		const allModels = this._ctx.getMirrorModels();
+		const models = allModels
+			// remove default libs
+			.filter((model) => !fileNameIsLib(model.uri))
+			// remove extra libs and previously removed libs
+			// Note that is is required for all model names to be unique, so
+			// while the logic here depends on the end user not creating extra
+			// libs with conflicting filenames, that is already a requirement.
+			.filter((model) => {
+				const uriAsString = model.uri.toString();
+				// if the extra lib was not given a name, then it gets an autogenerated name prefixed with ts:extralib-
+				if (uriAsString.startsWith('ts:extralib-')) {
+					return false;
+				}
+				// Otherwise, the prefix will be file:/// and the name will be what the user provided to add/setExtraLibs
```

Comment:
I don't think the prefix will always be `file://`. I think you can pass any URI (as a string) to `addExtraLib`, including with a different scheme, or with `file` as a scheme, but with a hostname (i.e. `file://some-host/some/path`.

Reply:
thanks for the catch @tomprince , sorry it took so long for me to revisit this.

I've made changes in 65058029d893ef15a829fc396b484c2b44038ca0 to address your comment